MessagingServiceTests/MessagingServicePOSTTests.cs

```diff
@@ -0,0 +1,975 @@
+            {
+                var startup = new Startup();
+                startup.ConfigureAADManager(aadManager);
+                startup.Configuration(app);
+            });
+            */
+
+        }
+
+        [ClassCleanup]
+        public static void ClassClean()
+        {
+            _server.Dispose();
+            _server_noStorage.Dispose();
+        }
+
+
+        [TestCleanup]
+        public async void Cleanup()
+        {
+            bool result = await _storageManager.Delete("email1");
+        }
+
+
+        /// <summary>
+        /// Tests happy path.
+        /// </summary>
+        [TestMethod]
+        public async Task TestContactOwners()
+        {
+            string fileContent = File.ReadAllText(TestJSONPath);
+            StringContent postContent = new StringContent(fileContent, Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await _server.HttpClient.PostAsync("/contactOwners", postContent);
+            Stream errors = response.Content.ReadAsStreamAsync().Result;
+
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                StreamReader errorsReader = new StreamReader(errors);
+                string errorsString = errorsReader.ReadToEnd();
+                JObject errorsJSON = JObject.Parse(errorsString);
+            }
+
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+
+
+            // Check message
+            StreamStorageContent messageJSON = (StreamStorageContent)await _storageManager.Load("email1");
+            StreamReader reader = new StreamReader(messageJSON.GetContentStream());
+            string bodyContent = reader.ReadToEnd();
+            JObject root = JObject.Parse(bodyContent);
+
+            Assert.AreEqual("user1@gmail.com,user2@gmail.com,user3@gmail.com", root["to"]);
+            Assert.AreEqual("someuser@live.com", root["from"]);
+            Assert.AreEqual("someuser@live.com", root["cc"]);
+            Assert.AreEqual("[PowerShellGallery] Message for owners of the module 'SomeTestPackage'", root["subject"]);
+            Assert.AreEqual(@"User rebro-1 &lt;someuser@live.com&gt; sends the following message to the owners of module 'SomeTestPackage':
+            
+            Hello owners, I would like to be an owner too.  Please add me!
+
+    -----------------------------------------------
+    To stop receiving contact emails as an owner of this module, sign in to the PowerShell Gallery and 
+    [change your email notification settings](/profile/edit).", root["body"]["text"]);
+            Assert.AreEqual(@"User rebro-1 &lt;someuser@live.com&gt; sends the following message to the owners of module 'SomeTestPackage':
+            
+            Hello owners, I would like to be an owner too.  Please add me!
+
+    -----------------------------------------------
+    <em>
+    To stop receiving contact emails as an owner of this module, sign in to the PowerShell Gallery and 
```

Comment:
I just learned about Markdown today, it is a cool markup language!  I agree, the plain text body should not have any of the Markdown characters.  I will change those.
For the HTML body, I think the Markdown is ok to keep if we can find a way around the URLs issue.  I'll look into if this is possible.